### ISSUE TITLE
CI/CD: Split building and tests

### ISSUE THREAD

**Issue by tlimoncelli** [StackExchange/dnscontrol#3257]
**Describe the bug**

`.github/workflows/pr_test.yml` builds the binaries (via GoReleaser) then runs the integration tests.  The integration tests do not use any artifacts of GoReleaser.

We should either split these into 2 parts so they can run in parallel OR not run GoReleaser at all.
